Fix serialization. Donation & Request relationship.

Everod52 opened the pull request:
Added relationship to Donation with request

Everod52 committed:
Fixed to_dict problem.
Added relationship to Donation with request

jaortiz117 reviewed (changes requested):
Voy a hacer los cambios al donations dao

Everod52 commented:
> Voy a hacer los cambios al donations dao

No hay problema.

jaortiz117 committed:
added .flaskenv

Everod52 commented on dao/mixin.py:
```diff
@@ -20,16 +20,17 @@ def to_dict(self, rel=None, backref=None):
-                    res[relation.key] = [i.to_dict(backref=self.__table__)
-                                         for i in value]
+                res[relation.key] = getattr(self, attr)
+                # if backref and backref == relation.table:
+                #     continue
+                # value = getattr(self, attr)
+                # if value is None:
+                #     res[relation.key] = None
+                # elif isinstance(value.__class__, DeclarativeMeta):
+                #     res[relation.key] = value.to_dict(backref=self.__table__)
+                # else:
+                #     res[relation.key] = [i.to_dict(backref=self.__table__)
+                #                          for i in value]
```
Delete commented code and TODO

Everod52 commented on dao/request.py:
```diff
@@ -1,15 +1,19 @@
-class Request(db.Model):
+class Request(OutputMixin, db.Model):
+    RELATIONSHIPS_TO_DICT = True
+
```
Add REQUEST_REQUIRED_PARAMS list

jaortiz117 committed:
added create and getter methods

jaortiz117 committed:
fixed create on request

jaortiz117 committed:
added update and delete methods

Ale573 reviewed (changes requested):
When you add a donation to the system, the gets doesn't work for the user. There is a problem with the dictionary.

yetsielaviles committed:
Added update and delete method to Request Entity.

Everod52 committed:
Fixed to_dict problem.
 Fixed relationship to Donation with request.
 Updated setUpDb script to drop all existing tables first.

Everod52 committed:
Removed unnecessary required params.
Created init methods.
Created query to get all objects for a specified relation for every user.

Everod52 committed:
Changed methods to static methods in UserHandler and User model.
Added query parameter for searching the donations and requests related to an user.

Everod52 committed:
Added Donation handler and routes.
Fixed a bug that verify_params method had that it didn't consider 0 values.
Added further functionality to routes using query parameters.
Added get all available donations.

Ale573 reviewed (approved):
Everything is working. All the gets, posts, updates, and deletes for users and donations.

Everod52 committed:
Removed print that was used for debugging.

yetsielaviles committed:
Added method to acquire requests and donations by user using user id.

Everod52 committed:
Updates create Request method.

Everod52 committed:
Merge branch 'r&d_relation' of https://github.com/uprm-inso-4115-2019-2020-S2/semester-project-inso4115-DisasterAid into r&d_relation

# Conflicts:
#	dao/request.py

Everod52 committed:
Implemented method to get the city of the user related to a donation.
Added query param to include city as a property.
Fixed some typos.
Delete unused imports in User.

Everod52 committed:
Reverted imports in User model. SqlAlchemy needs them.
Deleted prints.

Everod52 committed:
Implemented logic to prevent the creation of users with the same username.
